From: Cliff Wickman <cpw@sgi.com>
Subject: x86: SGI UV BAU initialization
References: bnc#561939, fate#306952
Patch-upstream: Yes

Commit e38e2af1c57c3eb5211331a5b4fcaae0c4a2a918 upstream.

A memory mapped register that affects the SGI UV Broadcast Assist Unit's
interrupt handling may sometimes be unintialized.

Remove the condition on its initialization, as that condition can be
randomly satisfied by a hardware reset.

Diffed against 2.6.32-rc7

Signed-off-by: Cliff Wickman <cpw@sgi.com>
Signed-off-by: Rafael J. Wysocki <rjw@suse.de>
---
 arch/x86/kernel/tlb_uv.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

Index: linux-2.6.32-master/arch/x86/kernel/tlb_uv.c
===================================================================
--- linux-2.6.32-master.orig/arch/x86/kernel/tlb_uv.c
+++ linux-2.6.32-master/arch/x86/kernel/tlb_uv.c
@@ -817,10 +817,8 @@ static int __init uv_init_blade(int blad
 	 */
 	apicid = blade_to_first_apicid(blade);
 	pa = uv_read_global_mmr64(pnode, UVH_BAU_DATA_CONFIG);
-	if ((pa & 0xff) != UV_BAU_MESSAGE) {
-		uv_write_global_mmr64(pnode, UVH_BAU_DATA_CONFIG,
+	uv_write_global_mmr64(pnode, UVH_BAU_DATA_CONFIG,
 				      ((apicid << 32) | UV_BAU_MESSAGE));
-	}
 	return 0;
 }
 
